examples/README.md

```diff
@@ -0,0 +1,55 @@
+# Examples
+
+This example is composed of two different but equal ways for composing two example components together
+* using the `wac` CLI
+* using the `wac-graph` library crate
+
+The example uses two input components (both located in the `deps` directory) that will be composed together:
+
+The `hello` component exports a function `hello` which returns a string:
+
+```bash
+# Print the wit for the hello component
+$ wasm-tools component wit deps/example/hello.wasm 
+package root:component;
+
+world root {
+  export hello: func() -> string;
+}
+```
+
+The `hello` exported function from `hello.wasm` will be plugged into the `hello` import of the `greeter` component which has the same signature as the `hello` exported function: 
+
+```bash
+# Print the wit for the greeter component
+$ wasm-tools component wit deps/example/greeter.wasm 
+package root:component;
+
+world root {
+  import hello: func() -> string;
+
+  export greet: func() -> string;
+}
+```
+
+The resulting composed component will therefore only have the exported `greet` function originally from the `greeter` component.
+
+## CLI
+
+`wac` can be used as a CLI tool. The `wac encode` command takes a wac script as input which defines how two components are composed together.
+
+Running the following command should produce a new component that is the composition of the `hello` and `greeter` components.
+
+```
+wac encode composition.wac -o composed.wasm  
+```
+
+*Note*: `wac encode` expects to find any input components inside of a `deps` folder in a directory named after the namespace part of the input component's name. In our example, the wac script uses the `example:greeter` and `example:hello` input components so `wac encode` expects to find those components in the `deps/example` directory.
```

Comment:
Is it worth mentioning here in the note that where `wac` looks for `deps/` is configurable through the `--deps-dir` flag?